src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs

```diff
@@ -306,7 +306,8 @@ void UpdateWhenChanged (string path, Action<Stream> generator)
 			var np  = path + ".new";
 			using (var o = File.OpenWrite (np))
 				generator (o);
-			Files.CopyIfChanged (np, path);
+			MonoAndroidHelper.CopyIfChanged (np, path);
+			MonoAndroidHelper.SetLastAccessAndWriteTimeUtc (path, DateTime.UtcNow, Log);
```

Comment:
@dellis1972 I was looking at this code. Do we really need this `.new` file that we delete afterward?

Can we just write to `path`, update the timestamp, and that's it?

Reply:
We do this allot. If we generate code we don't write directly to the file, we instead use a temp file and use 'CopyIfChanged' this ensures that we don't update files that we don't need to. That said in this case we are setting the date time.. so that kinda makes it redundant. 

Might be worth testing to see what impact it has.

Reply:
Ok, I think using the `.new` file might be right.

Consider:
- `generator(o)` writes to the original file and fails with some error
- We could have a 1/2 complete file with a newer timestamp, and the target wouldn't run again